File: TpWeb-main/angular/scholarIA/src/app/services/chat-message.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface ChatMessageCreationRequest {
  content: string;
  sessionId: string;
  [key: string]: any;
}

export interface ChatMessageUpdateRequest {
  content?: string;
  [key: string]: any;
}

export interface ChatMessageResponse {
  id: string;
  content: string;
  sessionId: string;
  userId?: string;
  createdAt?: string;
  updatedAt?: string;
  [key: string]: any;
}

@Injectable({
  providedIn: 'root'
})
export class ChatMessageService {
  private apiUrl = 'http://localhost:8080/api/v1/chat_message';

  constructor(private http: HttpClient) { }

  /**
   * Add a new chat message
   */
  addChatMessage(request: ChatMessageCreationRequest): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/add`, request);
  }

  /**
   * Update a chat message
   */
  updateChatMessage(chatMessageId: string, request: ChatMessageUpdateRequest): Observable<void> {
    return this.http.patch<void>(`${this.apiUrl}/${chatMessageId}`, request);
  }

  /**
   * Get all messages by session ID
   */
  getAllChatMessagesBySessionId(sessionId: string): Observable<ChatMessageResponse[]> {
    return this.http.get<ChatMessageResponse[]>(`${this.apiUrl}/session/${sessionId}`);
  }

  /**
   * Get chat message by ID
   */
  getChatMessageById(chatMessageId: string): Observable<ChatMessageResponse> {
    return this.http.get<ChatMessageResponse>(`${this.apiUrl}/${chatMessageId}`);
  }
}

File: TpWeb-main/back/src/main/java/com/research/paper/service/llm/ChatMessageService.java
package com.research.paper.service.llm;

import com.research.paper.dto.request.llm.chatMessage.ChatMessageCreation;
import com.research.paper.dto.request.llm.chatMessage.ChatMessageUpdate;
import com.research.paper.dto.response.llm.ChatMessageResponse;


import java.util.List;

public interface ChatMessageService {
    void addChatMessage(ChatMessageCreation request);
    void updateChatMessage(ChatMessageUpdate request, String chatMessageId);
    void deleteChatMessage(String chatMessageId);
    List<ChatMessageResponse> getAllChatMessagesBySessionId(String sessionId);
    ChatMessageResponse getChatMessageById(String chatMessageId);
}
